src/main/java/frc/robot/subsystems/colorwheel/commands/RotationControl.java

```diff
@@ -0,0 +1,57 @@
+package frc.robot.subsystems.colorwheel.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.Constants;
+import frc.robot.Constants.ColorWheel.*;
+import frc.robot.subsystems.colorwheel.ColorWheel;
+
+public class RotationControl extends CommandBase {
+
+    private ColorWheel colorWheel;
+    private String startingColor = "";
+    private String lastColor = "";
+    private String currentColor = "";
+    private double counter = 0;
+
+
+    public RotationControl(ColorWheel colorWheel) {
+        this.colorWheel = colorWheel;
+        addRequirements(colorWheel);
+    }
+
+    @Override
+    public void initialize() {
+        startingColor = colorWheel.whatColor();
+        lastColor = colorWheel.whatColor();
+
+    }
+
+    @Override
+    public void execute() {
+        // counting rotations method
+        colorWheel.setPower(Constants.ColorWheel.power);
+        currentColor = colorWheel.whatColor();
+        if (!currentColor.equals(lastColor)) {
+            lastColor = currentColor;
+            if (currentColor.equals(startingColor)) {
+                counter += 0.5;
+            }

```

Comment:
Why are incrementing in only 0.5? In `isFinished` you check whether the `counter >= 3`. This means you'll stop only when the Control Panel reaches six full rotations, and that's too high.

Reply:
In every rotation the same color appears twice.

![image](https://user-images.githubusercontent.com/42340005/133446791-224012ba-1154-4198-94a3-c1c47f394ace.png)